### PR TITLE
Myrec 3343 add identity token

### PR DESCRIPTION
**Pull request by diedhc01**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -38,7 +38,8 @@ dependencies {
     compile 'ch.qos.logback:logback-core:1.2.3'
     implementation 'com.amazonaws:aws-java-sdk-dynamodb'
     implementation 'com.amazonaws:aws-java-sdk-ssm'
-    implementation 'uk.co.bbc:unirecs-common:6.4.0'
+//    implementation 'uk.co.bbc:unirecs-common:6.4.0'
+    implementation 'uk.co.bbc:unirecs-common:6.5.0-milestone.0.0+20191001T161555Z'
```
Comment (SamGordonSmith) [inline build.gradle]: We can probably assume the next version will be 6.5.0 and drop the milestone part

File: src/test/java/uk/co/bbc/unirecs/chassis/dao/MockRuleSetDao.java [hunk 2]
```diff
@@ -34,7 +35,8 @@
     public static Rule getTestMatchingRule() {
         final Rule rule = new Rule();
         final RuleCondition ruleCondition1 = new RuleCondition();
-        ruleCondition1.setAttributeName("contentContext.contentId");
+//        ruleCondition1.setAttributeName("contentContext.contentId");
```
Comment (SamGordonSmith) [inline src/test/java/uk/co/bbc/unirecs/chassis/dao/MockRuleSetDao.java]: Can remove commented line

File: src/test/java/uk/co/bbc/unirecs/chassis/dao/MockRuleSetDao.java [hunk 3]
```diff
@@ -50,7 +52,8 @@ public static Rule getTestMatchingRule() {
     public static Rule getTestMatchingRule2() {
         final Rule rule = new Rule();
         final RuleCondition ruleCondition1 = new RuleCondition();
-        ruleCondition1.setAttributeName("contentContext.contentId");
+//        ruleCondition1.setAttributeName("contentContext.contentId");
```
Comment (SamGordonSmith) [inline src/test/java/uk/co/bbc/unirecs/chassis/dao/MockRuleSetDao.java]: Can remove commented line

File: src/main/java/uk/co/bbc/unirecs/chassis/controller/RecommendationsController.java [hunk 4]
```diff
@@ -176,13 +181,17 @@ public RecommendationsController(EnvironmentProperties environmentProperties,
...
-                                                              String contentId, String segmentId) {
+                                                              String contentId, String segmentId,
+                                                              String identityToken) {
         if (parameters.getContentContext() != null) {
             parameters.getContentContext().setContentId(contentId);
         }
         if (parameters.getUserContext() != null) {
             parameters.getUserContext().setSegmentId(segmentId);
         }
+        if (parameters.getContentContext() != null) {
```
Comment (kishorechimbili) [inline src/main/java/uk/co/bbc/unirecs/chassis/controller/RecommendationsController.java]: **if (parameters.getContentContext() != null)** is there in the top. so we can move **parameters.getUserContext().setIdentityToken(identityToken);** to that block

**Pull request merged**